client/utils/components.js

```diff
@@ -0,0 +1,53 @@
+import {settings} from './env'
+
+/**
+ * Calculate number of elements in array less than or equal to target values
+ *
+ * @param  {Number[]} arr     A sorted array of numbers
+ * @param  {[Number]} target  The number to be less than or equal to
+ * @return {int}              The number of elements in the array less than or equal to the target value
+ */
+export function calcNumLessThan (arr, target) {
```

Comment:
Just trying to watch out for instances of premature optimization. The simplest working solution is usually the best working solution.